File: 백준/Silver/1931. 회의실 배정/회의실 배정.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n')

const N = Number(input[0])
const targets = input.slice(1).map((row)=>row.split(' ').map(Number))

targets.sort((a,b)=> {
    if(a[1] === b[1]){
        return a[0]-b[0]
    }
    return a[1]-b[1]
})

let answer = 0
let end = -1

for(const [a,b] of targets){

    if(a >= end){
        answer +=1
        end = b
    }
}

console.log(answer)

File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
import sys
input = sys.stdin.readline
n = int(input().rstrip())
time = [list(map(int,input().split())) for _ in range(n)]

time.sort(key=lambda x:(x[1], x[0]))
answer = 0
end_time = -1

for a,b in time:
  if a>=end_time:
    answer +=1
    end_time = b

print(answer)
